src/v/cloud_storage/tests/cache_test.cc

```diff
@@ -60,22 +47,11 @@ FIXTURE_TEST(get_after_put, cache_test_fixture) {
 }
 
 FIXTURE_TEST(put_rewrites_file, cache_test_fixture) {
-    ss::sstring data_string1, data_string2;
-    while (data_string1.length() < 4096) {
-        data_string1 += SAMPLE_STRING1;
-    }
-    while (data_string2.length() < 4096) {
-        data_string2 += SAMPLE_STRING2;
-    }
-    iobuf buf1, buf2;
-    buf1.append(data_string1.data(), data_string1.length());
-    buf2.append(data_string2.data(), data_string2.length());
-
-    auto input1 = make_iobuf_input_stream(std::move(buf1));
-    cache_service.put(KEY, input1).get();
-
-    auto input2 = make_iobuf_input_stream(std::move(buf2));
-    cache_service.put(KEY, input2).get();
+    auto data_string1 = create_data_string(SAMPLE_STRING1, 4_KiB);
```

Comment:
Unrelated, but it is probably a good idea to go with bigger test strings (bigger than the page size and bigger than the iostream buffer, something like 1_MiB, but with a size not aligned to the page boundary). Sure they can't be easily compared visually, but the tests are more realistic this way (and I hope boost::test won't print them out in case of failure :)).

Reply:
Done.